---
title: "Discussion"
author: "Jonathan Speh"
format:
  docx:
    toc: false
    reference-doc: report_ref.docx
  html: 
    toc: false
  pdf: 
    toc: false
    pdf-engine: pdflatex
engine: knitr
execute: 
  cache: true
  echo: false
  output: true
bibliography: references.bib
---

```{r}
#| label: setup

library(knitr)
library(flextable)
options(scipen = 10000)
```

# Discussion

The project revealed an overlap between the deleterious CADD SNVs and the GnomAD Exom data of only about 10%. Furthermore, the observed variants were almost exclusively singletons or rare variants. These findings show that the variants that CADD predicts to be highly deleterious tend to be under purifying selection and are thus rare (or potentially absent) in the population which indicates good performance in predicting highly deleterious SNVs. However, a small number of common variants, i.e. variants not under purifying selection was also found, indicating that CADD classified them as more deleterious than they are.

## Frequently Observed Variants with high CADD Phred Scores

Amongst the five variants with the highest allele frequencies, two variants (rs6661174 on *FMO2* and rs601338 on *FUT2*) had clear molecular effects described in the relevant literature as they both lead to loss-of-function of their respective genes. Since these variants persist at high frequencies they cannot be highly deleterious. In the case of rs6661174, the reference allele is almost exclusively observed in sub-Saharan Africa while all other ancestry groups are homozygous for the alternative allele. @veeramah2008 explain this with the emergence of the variant before the migration of humans out of Africa, an evolutionary bottleneck that is responsible for the large genetic diversity in Africa compared to all other populations [@campbell2008]. The persistence of the variant since then indicates that it is at least evolutionary neutral. For rs601338, both alleles appear frequently in most populations. The variant is one of several common variants that lead to *FUT2* loss of function [@ferrer-admetlla2009]. As mentioned before, the *FUT2* genotype affects the ability to secrete AB0 Antibodies and is consequently relevant for immune defence against pathogens. While it is known that the secretor phenotype has a relevant impact on host-pathogen interactions, the consequence of this interaction is pathogen-specific and can depending on the pathogen be either beneficial or disadvantageous [@mottram2017; @wacklin2011]. A study by @ferrer-admetlla2009 highlights, that due to the strong evolutionary pressure of pathogen-host interaction, SNVs on *FUT2* such as the one that was observed here, are often under balancing selection. Similarly, @tennessen2012 observed that genes responsible for immune function are amongst the genes with the highest nucleotide diversity.

The SNV rs2285943 is located on *DNAH11,* a gene that in contrast to the aforementioned ones, has a clear association with a mendelian disease. Homozygotes and compound heterozygotes with non-functional *DNAH11* are affected by Primary Ciliary Dyskinesia Type 7 (CILD7) \[OMIM accession: #611884\] [@bartoloni2002; @lai2016; @schwabe2008]. However, rs2285943 is not described as a causal variant for CILD7 and it is annotated as benign in ClinVar. This might be explained by an alternative promoter on *DNAH11* that is listed in the UCSC Genome Browser [@raney2024]. The promoter is located on position chr7:21543346-21543495 (GRCh38), which means it is just downstream of the variant described here.

The last of the true stop gains that were analysed here was rs34358. The variant, which is located on *ANKDD1B,* is associated with both disease risk and protection. Unlike the aforementioned variant on *FUT2* which affects susceptibility to infectious diseases, rs34358 is associated with conditions such as migraine, type 2 diabetes and dyslipidemia [@vujkovic2020; @zhang2022]. These are however common disorders that appear later in life which makes them unlikely to be responsible for strong selective pressure [@blekhman2008]. Furthermore, the UCSC Genome Browser indicates that the variant is located on a cassette exon, i.e. an exon that is not retained in all transcripts, which might further reduce the biological effect of the variant.

The database and literature analysis of the variants with high allele frequencies revealed several potential mechanisms that could be responsible for the misclassification of common (thus likely neutral) variants as deleterious. The most obvious mechanism here was the incorrect Ensembl gene annotation seen at rs1147990. Since stop gains tend to be the most deleterious SNVs, the annotation as stop gain has a large influence on CADD's prediction. A similar mechanism might have influenced the high scores of rs2285943 and rs34358. While these SNVs are stop gains on protein-coding genes, the alternative promoter and the cassette exon might reduced their biological effects and reduce their deleteriousness. The probably most important explanation for missclassifications is however, that CADD cannot (and is not designed to) predict the pathogenicity of a variant. At least two of the five selected variants (rs6661174 and rs601338) have clear molecular effects and lead to loss of function of their respective genes. However, in both cases, the gene products are not essential for survival and reproduction and thus not under purifying selection.

## Enrichment Analysis of Observed and Simulated SNVs

To get an understanding of the differences between observed and only simulated variants on a larger scale, Gene Ontology and phenotype enrichment analyses have been performed. The analysis indicated that the gene list created from the simulated variants was enriched for several very low-level functions such as regulation of transcription and cell cycle control. In contrast, the genes on which SNVs could be observed were more often involved in somewhat higher functions such as metabolic processes or tissue organisation. Furthermore, the phenotype enrichment analysis showed, that genes with observed non-singleton SNVs are not as strongly associated with lethal phenotypes as those with only simulated or singleton variants. Taken together, the enrichment analysis indicates that SNVs on genes with basic regulatory functions tend to be more deleterious than those on genes involved in higher functions. A study of rare coding variants by @tennessen2012 has similarly found that genes in KEGG pathways that describe basic cellular functions have the lowest observable nucleotide diversity which indicates strong purifying selection at those sites and explains why variants are unlikely to be observed there.

## Relative cDNA Positions of Predicted Deleterious Variants

The last part of the project showed that SNVs with lower CADD Phred scores are located earlier in the genes than those with higher scores. Furthermore, it showed that SNVs with scores of at least 40 have an approximately normal distribution around a mean just before the midpoint of the genes. It has previously been observed that stop gains in the centre of a gene have a higher tendency to cause harm than those located closer to the start and to the end of genes, which corresponds well with the results here. However, except for the small plateau around a relative cDNA position of 0.25 for the observed dataset, no relevant differences between the observed and simulated variants could be seen. To increase understanding of the impact of cDNA position deleteriousness of a gene, a better control group, consisting for example of known and observed benign variants, is needed.

## Limitations

Several limitations should be considered in the interpretation of the results. Firstly, for the identification of observed SNVs, gnomAD exome data rather than whole genome data were used. This was based on the observation by @kircher2014, that SNVs with CADD Phred scores above 40 are exclusively found in the coding genome. However, this observation originates from the CADD publication, and the model has been updated since then. In the present study, a few variants in non-coding regions were also found to have CADD Phred scores above 40. These might have been misclassified as non-observed since exome datasets do not contain such variants. However, since the following analyses were focused on the functional implications of SNVs in coding genes, this is unlikely to have strongly impacted the results. A second aspect regarding the use of exome sequencing data needs to be discussed as well. Due to technical constraints of existing whole-exome sequencing (WES) methods, the coverage of the coding genome is somewhat incomplete and more biased than the coverage that can be achieved using whole-genome sequencing (WGS) [@barbitoff2020]. This could lead to existing variants not being detected by WES. On the other hand, gnomAD v4 contains about 10 times more sequenced exomes than genomes. Since the detection of rare variants depends on large sample sizes, the use of exome data in this project might have been beneficial in this aspect. For future work, it might be beneficial to integrate both WGS data for improved genome coverage and WES data to increase the ability to detect rare variants.

Another limitation is the way duplicated annotations were removed from the CADD dataset. This was done based only on the assumptions that the ConsScores and the relative position on the cDNA are good indicators for the gene model annotation that led to the highest CADD score for a variant. While this is likely true in most cases, it might not be the right approach in all situations and it was not sufficient to resolve all duplicates so random selection had to be used in these cases. A more robust way would have been to perform a *de novo* CADD scoring of simulated SNVs to select for each variant the annotation model that actually yielded the highest score. However, due to computational constraints, this was not feasible in the context of this project.

A potential limitation of the enrichment analysis is the way the gene lists were created. This was done by simply selecting genes on which at least one SNV with a score above 40 was observed more than once and contrasting them with genes on which only simulated variants were found. This approach does however not take into account that genes have different sizes and it treats all genes with observed variants equally, independently of how many predicted deleterious SNVs were observed on each of them. A way to account for this could have been to create a score that represents enrichment or depletion of observed SNVs on the genes and use a tool that allows the functional analysis of ranked gene lists such as gene set enrichment analysis described by @subramanian2005. For the phenotype enrichment analysis, an additional limitation is that due to a lack of suitable tools, no reduction of partially redundant terms was performed. This might result in overrepresentation of certain phenotypes such as those that have been described with both, complete and incomplete penetrance. Furthermore, the phenotype database used in this analysis was created using mouse models. While this might reduce the transferability of the results to humans, the use of animal models is currently the only way to perform to systematically study the effect of gene loss-of-function on the phenotype.

## Conclusion

Despite the aforementioned limitations, the project's results help to improve the understanding of SNVs that are predicted to be deleterious by CADD. They show that most of those variants are in fact under purifying selection and thus not or only rarely observed in the population. The project furthermore showed that especially genes involved in basic cellular processes are susceptible to deleterious variants and that many simulated SNVs that CADD predicts to be deleterious are located on such genes. However, the project also showed mechanisms that could lead CADD to overestimate the deleteriousness of variants. Taken together, this implies that the framework performs well in predicting highly deleterious SNVs, but its scoring might not be completely informative for individual variants and may need to be complemented by other systems that are more aimed at predicting pathogenicity.
