---
title: "Introduction"
author: "Jonathan Speh"
format:
  docx:
    df-print: paged 
    tbl-cap-location: top
engine: knitr
execute: 
  cache: true
  echo: false
  output: true
bibliography: references.bib
---

# Introduction

Since the introduction of Sanger sequencing, almost 50 years ago, the field of DNA sequencing has made enormous technical advances. Massive parallel DNA sequencing techniques such as Illumina can sequence a human genome within a few days, and more recent methods for long-read sequencing such as PacBio or Oxford Nanopore sequencing allow easier *de novo* assembly while increasing availability and portability of sequencing techniques [@shendure2017]. The relative technical ease of sequencing humane genomes stands opposed to the continuous challenge of interpreting genetic variation. Analyses of the 1000 Genomes Project revealed that an individual has - depending on their ancestry - between 4.1 and 5 million Variants compared to the reference genome. Most of these Variants were Single Nucleotide Variants (SNVs) or short insertions and deletions (indels), they tend to be located in the non-coding genome, and up to 200 000 Variants per individual had allele frequencies below 0.5% [@auton2015]. All of this illustrates that accurately distinguishing neutral (or even beneficial) variants from malignant variants that cause or increase risks for diseases is not a trivial task, which currently still limits the clinical utility of whole genome sequencing technologies [@lappalainen2021; @shendure2017]. Several methods to approach the challenge of predicting the effects of genetic variants have been developed. However, the use of such tools is often constrained by their limited scope on for example the coding genome [@brandes2023; @cheng2023], splice Sites [@jaganathan2019], or specific regulatory sites [@sample2019]. An alternative approach to the problem of estimating the effect of genetic variants is the Combined Annotation-Dependent Depletion (CADD) Framework, introduced by @kircher2014 . Instead of trying to predict direct molecular effects of genetic variants, CADD uses the predicted deleteriousness, I.e. an evolutionary measure of reduced fitness, of a variant as an indicator for pathogenicity. The method is based on the assumption, that alleles that can be observed at high frequencies (\>95%) in humans but don't exist in the human-chimpanzee ancestral genome are neutral or benign while simulated variants contain a significant number of deleterious variants that in reality would be removed by purifying selection. At the core of the CADD framework is a linear regression model, that was trained to use a large set of more than 100 annotations to distinguish the observed (proxy-neutral) from the simulated (proxy-deleterious) variants [@kircher2014]. Based on the annotations, the model computes a score (raw CADD score) that indicates how likely a variant belongs to the proxy-neutral or the proxy-deleterious class. After training, the model was applied to compute raw CADD scores for all theoretically possible SNVs in the human genome. To improve comparability of the results, the raw CADD score has then been scaled onto a Phred (-10 log~10~) scale from 0 to 99 where a score of 10 indicates a variant being amongst the top 10% with the highest predicted deleteriousness, while a score of 20 indicates the top 1% etc. [@kircher2014; @rentzsch2019]. Since its introduction, CADD has frequently been updated to incorporate the updated reference genome (GRCh38) and to improve its predictive performance [@rentzsch2019; @rentzsch2021; @schubach2024]. The model has been used in various research projects and has been found to be especially useful in prioritizing variants of unknown significance [@bongaerts2022; @tollefson2023].

Single Nucleotide Variants can have different molecular consequences. Depending on their location and the kind of substitution they can for example affect regulatory sites and splices sites or they can lead to changes in protein coding regions where they might result in loss or gain of stop codons or changes in the amino acid sequence. @kircher2014 found, that variants with a lower Phred score tend to be located in the non-coding regions of the genome, while variants with a higher score are more often located in protein-coding genes. At a Phred above 40, virtually all variants were stop-gains (nonsense mutations). Since stop-gain mutations often lead to a loss of function of a gene by nonsense-mediated mRNA decay or by synthesis of truncated proteins, this prediction appears to be biologically plausible. However, the SNVs that CADD predicts to be highly deleterious have not been further investigated to date. Thus, it is unknown which of these variants have been observed before and if they are known to be associated with phenotypes or diseases. This project aimed to address the questions of which of the SNVs with a Phred score of at least 40 have previously been observed and what distinguishes the observed variants from those that were only simulated. For this, the Genome Aggregation Database (GnomAD) [@karczewski2020] has been used to identify observed SNVs with a PHRED above 40. Identified variants with the highest allele frequencies have been further investigated using additional databases as well as literature research. Furthermore, gene ontology and phenotype enrichment analyses have been performed to compare genes that are affected by observed stop gains with those that are affected by the simulated stop gains. Lastly, the relative positions of stop-gains on their respective genes have been compared. The analysis showed that approximately 10% of the SNVs with Phred above 40 were found in GnomAD. The observed variants with high allele frequencies seemed to be neutral and simulated but not observed variants were more strongly associated with pathogenic effects than those that were observed. The results highlight that while CADD's use of deleteriousness as an indicator of pathogenicity is a good approach for prioritizing large sets of candidate variants, individual SNVs might be misclassified when using a defined threshold for binary classification of variants.
