---
title: "Report"
author: "Jonathan Speh"
format:
  docx:
    toc: false
    reference-doc: report_ref.docx
  html: 
    toc: false
  pdf: default
engine: knitr
execute: 
  cache: true
  echo: false
  output: true
bibliography: references.bib
---

```{r}
#| label: setup

library(knitr)
library(flextable)
options(scipen = 10000)
```

# Abstract

**Background:** Assessing the effect of genetic variants is currently the largest challenge in the application of next-generation sequencing in clinical settings. One approach to this challenge is the Combined Annotation-Dependent Depletion (CADD) framework which predicts the deleteriousness of variants based on a large number of annotations. Here, the 0.01% of all theoretically possible single nucleotide variants (SNVs) that CADD scored to be most deleterious were analysed to understand which of those variants were observed in the population and what distinguishes observed variants from those that were only simulated.

**Methods:** The gnomAD database was used to identify observed SNVs from the CADD-scored data. Literature research was performed for the SNVs with the highest allele frequency, and gene enrichment analysis was used for functional analysis of observed and only simulated variants. Furthermore, a comparison of the relative position of observed and simulated SNVs on their genes was made.

**Results:** Approximately 10% of SNVs that CADD predicts to be highly deleterious were observed in gnomAD. This included a few common variants which due to different mechanisms were misclassified as deleterious by CADD. Observed variants appear to be less strongly associated with basic cellular processes and lethal phenotypes than only simulated SNVs.

**Conclusion:** Most SNVs that CADD predicts to be highly deleterious appear not or only rarely in the population, but several observed cases where CADD overestimated the deleteriousness highlight that additional measures to directly estimate pathogenicity could improve the assessment of unknown variants.

# Introduction

Since the introduction of Sanger sequencing, almost 50 years ago, the field of DNA sequencing has made enormous technical advances. Massive parallel DNA sequencing techniques such as the Illumina plattforms can sequence a human genome within a few days, and more recent methods for long-read sequencing such as PacBio or Oxford Nanopore sequencing allow easier *de novo* assembly while increasing availability and portability of sequencing techniques [@shendure2017]. The relative technical ease of sequencing humane genomes stands opposed to the continuous challenge of interpreting genetic variation. Analyses of the 1000 Genomes Project revealed that an individual has - depending on their ancestry - between 4.1 and 5 million variants compared to the reference genome [@auton2015]. Most of these variants are Single Nucleotide Variants (SNVs) or short insertions and deletions (indels) located in the non-coding genome, and up to 200,000 Variants per individual have allele frequencies below 0.5% [@auton2015]. All of this illustrates that accurately distinguishing neutral (or even beneficial) variants from malignant variants that cause or increase risks for diseases is not a trivial task, still limiting the clinical utility of whole genome sequencing technologies [@lappalainen2021; @shendure2017]. Several methods to approach the challenge of predicting the effects of genetic variants have been developed. However, the use of such tools is often constrained by their limited scope on for example the coding genome [@brandes2023; @cheng2023], splice sites [@jaganathan2019], or specific regulatory sites [@sample2019]. An alternative approach to the problem of estimating the effect of genetic variants is the Combined Annotation-Dependent Depletion (CADD) Framework, introduced by @kircher2014 . Instead of trying to predict direct molecular effects of genetic variants, CADD tries to infer deleteriousness, an evolutionary measure of reduced fitness of a variant, as an indicator for pathogenicity. The method is based on the assumption, that alleles that can be observed at very high frequencies (\>95%) in humans or are fixed derived since human-chimpanzee ancestral genome (i.e. are shared among all present-day humans but changed in a human ancestor) are neutral or benign while simulated variants contain a significant number of deleterious variants that would typically be removed by generations of purifying selection. At the core of the CADD framework is a linear regression model, that was trained to use a large set of more than 100 annotations to distinguish the observed (proxy-neutral) from the simulated (proxy-deleterious) variants [@kircher2014]. Based on the annotations, the model computes a score (raw CADD score) that indicates how likely a variant belongs to the proxy-neutral or the proxy-deleterious class. After training, the model is applied to compute raw CADD scores for all theoretically possible SNVs in the human genome. To improve comparability of the results, the raw CADD score is then been scaled onto a Phred (-10 log~10~) scale from 0 to 99 where a score of 10 indicates a variant being amongst the top 10% with the highest predicted deleteriousness, while a score of 20 indicates the top 1% etc. [@kircher2014; @rentzsch2019]. Since its introduction, CADD has frequently been updated to incorporate the updated reference genome (GRCh38) and to improve its predictive performance [@rentzsch2019; @rentzsch2021; @schubach2024]. The model has been used in various research projects and has been found to be especially useful in prioritizing variants of unknown significance [@bongaerts2022; @tollefson2023].

SNVs can have different molecular consequences. Depending on their location and the kind of substitution they can for example affect regulatory sites and splice sites or they can lead to changes in protein coding regions where they might result in loss or gain of stop codons or changes in the amino acid sequence. @kircher2014 found that variants with a lower CADD Phred score tend to be located in the non-coding regions of the genome, while variants with a higher score are more often located in protein-coding genes. At a CADD Phred above 40, virtually all variants were stop-gains (nonsense mutations). Since stop-gain mutations often lead to a loss of function of a gene by nonsense-mediated mRNA decay or by synthesis of truncated proteins, this prediction appears to be biologically plausible. However, the SNVs that CADD predicts to be highly deleterious have not been further investigated to date. Thus, it is unknown which of these variants have been observed before and if they are known to be associated with phenotypes or diseases. This project addresses the questions of which of the SNVs with a Phred score of at least 40 have previously been observed and what distinguishes the observed variants from those that were only simulated. For this, the Genome Aggregation Database (gnomAD) [@karczewski2020] has been used to identify observed SNVs with a PHRED above 40. Identified variants with the highest allele frequencies have been further investigated using additional databases as well as literature research. Furthermore, gene ontology and phenotype enrichment analyses have been performed to compare genes that are affected by observed stop gains with those that are affected by the simulated stop gains. Lastly, the relative positions of stop-gains on their respective genes have been compared. The analysis showed that approximately 10% of the SNVs with Phred above 40 were found in gnomAD. The observed variants with high allele frequencies seemed to be neutral and simulated but not observed variants were more strongly associated with pathogenic effects than those that were observed. The results highlight that while CADD's use of deleteriousness as an indicator of pathogenicity is a good approach for prioritizing large sets of candidate variants, individual SNVs might be misclassified when using a defined threshold for binary classification of variants.

# Methods

## Identification of Observed SNVs

To identify SNVs with a CADD Phred above 40 observed in the gnomAD database, CADD v1.6 pre-scored variants \[links available here: <https://cadd.bihealth.org/download>\] were filtered for a CADD Phred score of greater or equal than 40. GnomAD release 4.0 exome data were retrieved from <https://gnomAD.broadinstitute.org/downloads>. Variants that did not pass die gnomAD quality control filters were removed and only SNVs retained.

For each variant, a unique identifier consisting of chromosome, position, reference allele, and alternative allele was generated. This identifier was then used to identify observed variants by computing the overlap between the two datasets so that all CADD variants also found in the gnomAD dataset were retained. Only variants found on the autosomes were considered for the subsequent analyses.

Filtering of the CADD and gnomAD datasets and generation of the overlap has been performed using the BASH command line interpreter and GNU tools. Subsequent data manipulation and analysis were performed using R version 4.2.2 [@base] and the packages data.table version 1.14.8 [@data.table] as well as dplyr version 1.1.4, tidyr version 1.3.0, stringr version 1.5.1 and ggplot2 version 3.4.4 provided by the tidyverse metapackage [@tidyverse].

Variants can be annotated using different gene models leading to duplicated variants that are differentially annotated in the CADD dataset. However, the CADD output contains only the highest score for each variant without indicating which annotation the score is based on. To account for this, a three-step filtering strategy was used to ensure that only unique variants were retained. Firstly, for each variant, the annotation with the highest ConsScore was prioritized. The ConScore is a score from 0 to 8 which is based on the assumption that some types of consequences (such as stop-gain, non-synonymous base exchange, variants in splice sites) tend to be more deleterious than others (such as synonymous or intergenic exchanges) \[assignment of ConScore: <https://github.com/kircherlab/CADD-scripts/blob/a89404c4edf8b33bffc7d8f79fcd77dc5ac6496a/src/scripts/lib/Annotations.py#L175>\]. The score is returned as one of the CADD annotations, it is however not used in the model itself. Secondly, variants that occur earlier in a gene (have a smaller relative cDNA position) have been prioritized over those that appear later. Lastly, duplicates that could not be resolved by the aforementioned methods were resolved by random selection.

## Analysis of Variants with the Highest Allele Frequencies

Of the Variants observed in gnomAD, the five variants with the highest allele frequencies were selected for further literature research. Relevant literature that might explain the consequences of the variants was identified using the LitVar2 text mining tool [@allot2023] Furthermore, the ClinVar Database [@landrum2020] was used to search for the clinical significance of the variants, and OMIM [@hamosh2005] was used to investigate phenotypes/diseases that are associated with genes affected by the variants.

## Enrichment Analysis

Gene ontology and phenotype enrichment analysis have been performed with the enrichr gene set analysis tool [@xie2021] using the enrichR package [@enrichR]. For the Gene Ontology (GO) enrichment analysis, two gene lists have been created based on the variants with a CADD Phred of at least 40. One list contained genes that were observed more than once in gnomAD while the other one contained genes derived from simulated variants not found in gnomAD. For the phenotype enrichment analysis a third gene list was used. This list contained genes where singleton variants (i.e. variants observed on a single chromosome in a single individual in the gnomAD database) were found. The databases used were GO_Biological_Process_2023 and MGI_Mammalian_Phenotype_Level_4_202 for gene ontology and phenotype enrichment analysis respectively. Terms with a Benjamini-Hochberg corrected P-value of 0.05 or lower were considered significantly enriched. Redundant GO terms were reduced based on semantic similarities using the rrvgo package [@rrvgo] to first compute a similarity matrix of the GO terms and then to reduce the terms in the matrix using -log10 transformed adjusted p-value as weights and a threshold of 0.7.

## Analysis of Relative cDNA Positions

To identify where along the genes deleterious stop gains are most often located, the distribution of the relative complementary DNA (cDNA) position (i.e. the relative position along the mature RNA transcript) of observed and simulated stop-gains was plotted using the geom_density function provided by ggplot2 and the geom_density_ridges function provided by ggridges packages version 0.5.4 [@ggridges] using default parameters.

# Results

```{r}
#| include: false
#| cache: false
library(data.table)
library(dplyr)

all_phred40_snvs <- fread("processed_data/all_phred40.csv")
overlap_phred40 <- fread("processed_data/overlap_phred40_unique_parsed.csv")

n_phred40 <- nrow(all_phred40_snvs)
n_observed <- nrow(overlap_phred40)
```

A total of `r format(n_phred40, big.mark = ",")` variants on the autosomes had a CADD Phred score of at least 40. Of those variants, `r format(n_observed, big.mark = ","` (`r sprintf("%0.1f%%", n_observed/n_phred40 * 100)`) were found in the gnomAD database.

Almost all variants, ``` r``format(``nrow(filter(overlap_phred40, Consequence == "STOP_GAINED")), big.mark = ",") ``` were stop-gains, while only `r nrow(filter(overlap_phred40, Consequence == "NON_SYNONYMOUS"))` non-synonymous, `r nrow(filter(overlap_phred40, Consequence == "INTRONIC"))` intronic, five splice site variants, two regulatory, and one variant in the 3' UTR (untranslated region of mature RNA after protein coding sequence) were identified. The majority of the observed SNVs (`r sprintf("%0.1f%%", nrow(filter(overlap_phred40, AC == 1))/n_observed * 100)`) are singletons . Only `r format(nrow(filter(overlap_phred40, AF >= 0.01)), big.mark = ",")` variants had an allele frequency (AF) of at least 1% (i.e. are not considered rare variants). Interestingly, three variants with an AF greater than 50% were found in the observed data. Table 1 shows an overview of the identified SNVs with the highest allele frequencies.

::: {#tbl-one}
```{r}

overlap_phred40 |>
  slice_max(AF, n = 5) |>
  select(Chrom, Pos, Ref, Alt, "CADD Phred" = PHRED, "Gene" = GeneName, AF) |> 
#  kable(align = "lllllll") |>
  flextable() |>
  theme_booktabs(bold_header = TRUE) |>
  align(align = "left", part = "all") |>
  italic(j = "Gene") |> 
  flextable::colformat_num(j = 2, big.mark = "") |>
  
  #set_header_labels(Chrom = "Chromosome", Pos = "Position", 
  #          Ref = "Reference", Alt = "Alternative", PHRED = "PHRED", GeneName = "Gene", 
  #          AF = "Allele Frequency") |>
  fontsize(size = 10)

   

```

SNVs with a CADD Phred score above 40 that had the highest allele frequencies in gnomAD.
:::

## Literature Research of Variants with high Allele Frequencies

### 1-171208951-C\>T (rs6661174)

The observed SNV with the highest allele frequency was rs6661174. It leads to a premature stop codon on exon 9 of *FMO2*, a gene coding for Flavin Containing Dimethylaniline Monooxygenase 2. The alternative allele (T-allele) has an overall AF of close to one. It has allele frequencies of or close to 1 (99.52%) throughout all gnomAD ancestry groups except African / African American where the frequency is only 0.88. The T allele leads to *FMO2* loss of function while the ancestral C allele leads to a functional allele. A study by @veeramah2008 found high frequencies of the C allele in the population of sub-Saharan Africa where approximately one-third was at least heterozygote, but could not find evidence for selective pressure on either allele. In contrast, @mekonnen2017 propose that in specific African populations, the persistence of the functional allele can be explained by its association with reduced Tuberculosis disease activity.

### 5-75669297-G\>A (rs34358)

This variant is a stop-gain in *ANKDD1B*, coding for Ankyrin Repeat And Death Domain Containing 1B. The alternative A allele is the major allele in all gnomAD ancestry groups except South Asian and East Asian where it has AFs of 0.44 and 0.35 respectively. The variant has been described as a risk allele for type 2 diabetes and dyslipidemia [@vujkovic2020] and as a protective allele for migraine without aura [@zhang2022], however no studies investigating the evolutionary history of the allele and no evidence for a deleterious effect (like a pathogenic effect with early onset and high penetrance) could be found.

### 1-54716627-T\>A (rs1147990)

This variant has an overall frequency of 0.55. It is most frequent in the East Asian (0.98) and South Asian (0.72) ancestry groups. Although being annotated as stop gain in *MROH7-TTC4* by CADD v1.6, the variant is located in a read-trough transcript between *MROH7* and *TTC4,* two neighboring genes. Multiple variants of the transcript can arise from alternative splicing, however, they are likely degraded by nonsense-mediated decay and do not encode any protein. In CADD v1.7 which uses updated Ensembl gene annotations, the transcript is no longer annotated as a coding gene. With that change, the variant's Phred score was markedly reduced to 19.97.

### 7-21543345-G\>T (rs2285943)

Rs2285943 is predicted to result in a stop gain on exon 1/82 of *DNAH11.* The variant is common (AFs 0.29 - 0.55) throughout all gnomAD ancestry groups. It is known that DNAH11 loss of function leads to Primary Ciliary Dyskineasia [@bartoloni2002; @lai2016; @schwabe2008], however, multiple ClinVar submissions indicate that rs2285943 is not linked to that condition.

### 19-48703417-G\>A (rs601338)

The fifth of the selected variants leads to a loss of function in *FUT2.* With allele frequencies between 0.30 in Admixed Americans and 0.52 in Middle Easterns, it is common in most ancestry groups. However, the variant is rare in East Asians (AF 0.002). The variant affects the function of the alpha(1,2)fucosyltransferase which is responsible for secretion of ABO antigens. Carriers of a functional *FUT2* allele secrete these antigens in their bodily fluids while homozygous or compound heterozygous carriers of non-functional FUT2 (for example caused by rs601338) do not secrete the antigens. The secretor phenotype (secretor polymorphism) seems to be associated as both, risk or protective factor, with numerous conditions and diseases [@krog2023; @mottram2017].

## Enrichment analysis

```{r}

GOBP_observed <- fread("processed_data/GOBP_observed.csv")
GOBP_observed_reduced <- fread("processed_data/GOBP_observed_reduced.csv")

GOBP_simulated <- fread("processed_data/GOBP_simulated.csv")
GOBP_simulated_reduced <- fread("processed_data/GOBP_simulated_reduced.csv")

PT_simulated <- fread("processed_data/pte_simulated.csv")
PT_singletons <- fread("processed_data/pte_singleton.csv")
PT_non_singletons <- fread("processed_data/pte_non_singleton.csv")
```

Gene ontology enrichment analysis revealed `r nrow(GOBP_observed)` significantly enriched biological processes for the genes that contained observed SNVs, resulting in `r max(GOBP_observed_reduced$cluster)` clusters with semantically distinct functions. Analysis of the gene list derived from simulated but not observed SNVs yielded `r nrow(GOBP_simulated)` enriched terms in `r max(GOBP_simulated_reduced$cluster)` distinct clusters. Figure 1 shows the enriched clusters for both groups.

::: {#fig-treemaps}
![](/img/Fig1.png)

Treemaps of the biological processes that are enriched in gene lists derived from observed (left) and simulated (right) variants. The treemaps reflect the hierarchical nature of gene ontology terms by grouping terms that were found to be semantically similar under a parent term. The size of the rectangles indicates the negative log10 adjusted p-value.
:::

Genotype enrichment analysis showed that genes with observed SNVs with a CADD Phred score above 40 tend to be involved in processes that affect cell and tissue morphology (extracellular structure organization and intermediate filament organisation, cell-matrix adhesion), cation transport, and metabolic processes (fatty acid metabolic process, proteolysis, glycerophospholipid biosynthetic processes). One the other hand, the gene list with only simulated SNVs was mainly enriched for processes that regulate gene expression, cell proliferation, and differentiation or are involved in cellular signaling.

The phenotype enrichment analysis revealed `r nrow(PT_simulated)`, `r nrow(PT_singletons)`, and `r nrow(PT_non_singletons)` significantly enriched terms for the gene lists derived from simulated, singleton and non-singleton SNVs, respectively. The 10 most significantly enriched terms (lowest adjusted P-value) are shown in Figure 2.

::: {#fig-pte}
![](/img/plot-pte-facet-1.png)

The ten most significantly enriched phenotypes for genes on which simulated, singleton, and non-singleton SNVs with CADD Phred scores above 40 are found. The color of the bars represents the statistical significance as negative log10 adjusted P-value and the bars show the fraction of genes found in the respective gene lists compared to all genes that map to the phenotype.
:::

The phenotype enrichment analysis showed, that in the simulated and singleton groups the most significantly enriched phenotypes lead to early (embryonic, pre- and postnatal or preweaning) lethality while the phenotypes in the non-singleton group appear less likely to lead to this early lethality. Furthermore, genes from simulated variants tend to have higher overlaps than those from singleton or non-singleton SNVs.

## Relative positions of simulated and observed SNVs

```{r}
observed_simulated <- fread("processed_data/observed_simulated.csv") |> filter(PHRED >= 40)

mean_relPos_obs <- mean(filter(observed_simulated, flag == "observed")$relcDNApos)

mean_relPos_sim <- mean(filter(observed_simulated, flag == "simulated")$relcDNApos)
```

To see where deleterious SNVs are located on the individual genes, the relative cDNA positions of observed and simulated stop-gains were plotted.

::: {#fig-positions}
![](/img/Fig3.png)

**(a)** Distribution of the stop-gains in different CADD Phred score groups. **(b)** Relative cDNA positions of stop-gains with Phred scores above 40 that have been observed (red) or simulated (blue). Relative cDNA position refers to the postion along the mature mRNA transcript that is being translated to a protein.
:::

Figure 3a shows that stop gains with a lower Phred score tend to have a lower relative cDNA position, meaning they are located closer to the start of a gene. In figure 3b it can be seen that the distribution of the observed variants roughly follows the distribution of the simulated ones. However, while the simulated variants appear to be normally distributed around their mean of `r round(mean_relPos_sim, 2)` while the observed variants have a slightly lower mean (`r round(mean_relPos_obs, 2)`) and a distribution with a small plateau around 0.25.

# Discussion

The project revealed an overlap between the deleterious CADD SNVs and the GnomAD Exom data of only about 10%. Furthermore, the observed variants were almost exclusively singletons or rare variants. These findings show that the variants that CADD predicts to be highly deleterious tend to be under purifying selection and are thus rare (or potentially absent) in the population which indicates good performance in predicting highly deleterious SNVs. However, a small number of common variants, i.e. variants not under purifying selection was also found, indicating that CADD classified them as more deleterious than they are.

## Frequently Observed Variants with high CADD Phred Scores

Amongst the five variants with the highest allele frequencies, two variants (rs6661174 on *FMO2* and rs601338 on *FUT2*) had clear molecular effects described in the relevant literature as they both lead to loss-of-function of their respective genes. Since these variants persist at high frequencies they cannot be highly deleterious. In the case of rs6661174, the reference allele is almost exclusively observed in sub-Saharan Africa while all other ancestry groups are homozygous for the alternative allele. @veeramah2008 explain this with the emergence of the variant before the migration of humans out of Africa, an evolutionary bottleneck that is responsible for the large genetic diversity in Africa compared to all other populations [@campbell2008]. The persistence of the variant since then indicates that it is at least evolutionary neutral. For rs601338, both alleles appear frequently in most populations. The variant is one of several common variants that lead to *FUT2* loss of function [@ferrer-admetlla2009]. As mentioned before, the *FUT2* genotype affects the ability to secrete AB0 Antibodies and is consequently relevant for immune defence against pathogens. While it is known that the secretor phenotype has a relevant impact on host-pathogen interactions, the consequence of this interaction is pathogen-specific and can depending on the pathogen be either beneficial or disadvantageous [@mottram2017; @wacklin2011]. A study by @ferrer-admetlla2009 highlights, that due to the strong evolutionary pressure of pathogen-host interaction, SNVs on *FUT2* such as the one that was observed here, are often under balancing selection. Similarly, @tennessen2012 observed that genes responsible for immune function are amongst the genes with the highest nucleotide diversity.

The SNV rs2285943 is located on *DNAH11,* a gene that in contrast to the aforementioned ones, has a clear association with a mendelian disease. Homozygotes and compound heterozygotes with non-functional *DNAH11* are affected by Primary Ciliary Dyskinesia Type 7 (CILD7) \[OMIM accession: #611884\] [@bartoloni2002; @lai2016; @schwabe2008]. However, rs2285943 is not described as a causal variant for CILD7 and it is annotated as benign in ClinVar. This might be explained by an alternative promoter on *DNAH11* that is listed in the UCSC Genome Browser [@raney2024]. The promoter is located on position chr7:21543346-21543495 (GRCh38), which means it is just downstream of the variant described here.

The last of the true stop gains that were analysed here was rs34358. The variant, which is located on *ANKDD1B,* is associated with both disease risk and protection. Unlike the aforementioned variant on *FUT2* which affects susceptibility to infectious diseases, rs34358 is associated with conditions such as migraine, type 2 diabetes and dyslipidemia [@vujkovic2020; @zhang2022]. These are however common disorders that appear later in life which makes them unlikely to be responsible for strong selective pressure [@blekhman2008]. Furthermore, the UCSC Genome Browser indicates that the variant is located on a cassette exon, i.e. an exon that is not retained in all transcripts, which might further reduce the biological effect of the variant.

The database and literature analysis of the variants with high allele frequencies revealed several potential mechanisms that could be responsible for the misclassification of common (thus likely neutral) variants as deleterious. The most obvious mechanism here was the incorrect Ensembl gene annotation seen at rs1147990. Since stop gains tend to be the most deleterious SNVs, the annotation as stop gain has a large influence on CADD's prediction. A similar mechanism might have influenced the high scores of rs2285943 and rs34358. While these SNVs are stop gains on protein-coding genes, the alternative promoter and the cassette exon might reduced their biological effects and reduce their deleteriousness. The probably most important explanation for missclassifications is however, that CADD cannot (and is not designed to) predict the pathogenicity of a variant. At least two of the five selected variants (rs6661174 and rs601338) have clear molecular effects and lead to loss of function of their respective genes. However, in both cases, the gene products are not essential for survival and reproduction and thus not under purifying selection.

## Enrichment Analysis of Observed and Simulated SNVs

To get an understanding of the differences between observed and only simulated variants on a larger scale, Gene Ontology and phenotype enrichment analyses have been performed. The analysis indicated that the gene list created from the simulated variants was enriched for several very low-level functions such as regulation of transcription and cell cycle control. In contrast, the genes on which SNVs could be observed were more often involved in somewhat higher functions such as metabolic processes or tissue organisation. Furthermore, the phenotype enrichment analysis showed, that genes with observed non-singleton SNVs are not as strongly associated with lethal phenotypes as those with only simulated or singleton variants. Taken together, the enrichment analysis indicates that SNVs on genes with basic regulatory functions tend to be more deleterious than those on genes involved in higher functions. A study of rare coding variants by @tennessen2012 has similarly found that genes in KEGG pathways that describe basic cellular functions have the lowest observable nucleotide diversity which indicates strong purifying selection at those sites and explains why variants are unlikely to be observed there.

## Relative cDNA Positions of Predicted Deleterious Variants

The last part of the project showed that SNVs with lower CADD Phred scores are located earlier in the genes than those with higher scores. Furthermore, it showed that SNVs with scores of at least 40 have an approximately normal distribution around a mean just before the midpoint of the genes. It has previously been observed that stop gains in the centre of a gene have a higher tendency to cause harm than those located closer to the start and to the end of genes, which corresponds well with the results here. However, except for the small plateau around a relative cDNA position of 0.25 for the observed dataset, no relevant differences between the observed and simulated variants could be seen. To increase understanding of the impact of cDNA position deleteriousness of a gene, a better control group, consisting for example of known and observed benign variants, is needed.

## Limitations

Several limitations should be considered in the interpretation of the results. Firstly, for the identification of observed SNVs, gnomAD exome data rather than whole genome data were used. This was based on the observation by @kircher2014, that SNVs with CADD Phred scores above 40 are exclusively found in the coding genome. However, this observation originates from the CADD publication, and the model has been updated since then. In the present study, a few variants in non-coding regions were also found to have CADD Phred scores above 40. These might have been misclassified as non-observed since exome datasets do not contain such variants. However, since the following analyses were focused on the functional implications of SNVs in coding genes, this is unlikely to have strongly impacted the results. A second aspect regarding the use of exome sequencing data needs to be discussed as well. Due to technical constraints of existing whole-exome sequencing (WES) methods, the coverage of the coding genome is somewhat incomplete and more biased than the coverage that can be achieved using whole-genome sequencing (WGS) [@barbitoff2020]. This could lead to existing variants not being detected by WES. On the other hand, gnomAD v4 contains about 10 times more sequenced exomes than genomes. Since the detection of rare variants depends on large sample sizes, the use of exome data in this project might have been beneficial in this aspect. For future work, it might be beneficial to integrate both WGS data for improved genome coverage and WES data to increase the ability to detect rare variants.

Another limitation is the way duplicated annotations were removed from the CADD dataset. This was done based only on the assumptions that the ConsScores and the relative position on the cDNA are good indicators for the gene model annotation that led to the highest CADD score for a variant. While this is likely true in most cases, it might not be the right approach in all situations and it was not sufficient to resolve all duplicates so random selection had to be used in these cases. A more robust way would have been to perform a *de novo* CADD scoring of simulated SNVs to select for each variant the annotation model that actually yielded the highest score. However, due to computational constraints, this was not feasible in the context of this project.

A potential limitation of the enrichment analysis is the way the gene lists were created. This was done by simply selecting genes on which at least one SNV with a score above 40 was observed more than once and contrasting them with genes on which only simulated variants were found. This approach does however not take into account that genes have different sizes and it treats all genes with observed variants equally, independently of how many predicted deleterious SNVs were observed on each of them. A way to account for this could have been to create a score that represents enrichment or depletion of observed SNVs on the genes and use a tool that allows the functional analysis of ranked gene lists such as gene set enrichment analysis described by @subramanian2005. For the phenotype enrichment analysis, an additional limitation is that due to a lack of suitable tools, no reduction of partially redundant terms was performed. This might result in overrepresentation of certain phenotypes such as those that have been described with both, complete and incomplete penetrance. Furthermore, the phenotype database used in this analysis was created using mouse models. While this might reduce the transferability of the results to humans, the use of animal models is currently the only way to perform to systematically study the effect of gene loss-of-function on the phenotype.

## Conclusion

Despite the aforementioned limitations, the project's results help to improve the understanding of SNVs that are predicted to be deleterious by CADD. They show that most of those variants are in fact under purifying selection and thus not or only rarely observed in the population. The project furthermore showed that especially genes involved in basic cellular processes are susceptible to deleterious variants and that many simulated SNVs that CADD predicts to be deleterious are located on such genes. However, the project also showed mechanisms that could lead CADD to overestimate the deleteriousness of variants. Taken together, this implies that the framework performs well in predicting highly deleterious SNVs, but its scoring might not be completely informative for individual variants and may need to be complemented by other systems that are more aimed at predicting pathogenicity.

# References
