---
title: "Results"
author: "Jonathan Speh"
format:
  docx:
    df-print: paged 
    tbl-cap-location: top
engine: knitr
execute: 
  cache: true
  echo: false
  output: true
bibliography: references.bib
---

```{r}
#| label: setup

library(knitr)
library(flextable)

```

```{r}
#| include: false
#| cache: false
library(data.table)
library(dplyr)

all_phred40_snvs <- fread("processed_data/all_phred40.csv")
overlap_phred40 <- fread("processed_data/overlap_phred40_unique_parsed.csv")

n_phred40 <- nrow(all_phred40_snvs)
n_observed <- nrow(overlap_phred40)

```

A total of `r n_phred40` variants on the autosomes had a PHRED score of at least 40. Of those variants, `r n_observed` (`r sprintf("%0.1f%%", n_observed/n_phred40 * 100)`) were found in the GnomAD database.

Almost all variants (`r nrow(filter(overlap_phred40, Consequence == "STOP_GAINED"))`) were stop-gains, while only `r nrow(filter(overlap_phred40, Consequence == "NON_SYNONYMOU"))` non synonymous, `r nrow(filter(overlap_phred40, Consequence == "INTRONIC"))`, intronic, `r nrow(filter(overlap_phred40, Consequence == "SPLICE_SITE"))` splice site, `r nrow(filter(overlap_phred40, Consequence == "REGULATORY"))` regulatory, and `r nrow(filter(overlap_phred40, Consequence == "3PRIME_UTR"))` one Variant in the 3' UTR were identified. The majority of the observed SNVs (`r sprintf("%0.1f%%", nrow(filter(overlap_phred40, AC == 1))/n_observed * 100)`) are singletons (I.e. they have only been observed once). Only `r nrow(filter(overlap_phred40, AF >= 0.01))` variants had an allele frequency (AF) of at least 1% (i.e. are not rare variants). Interestingly, three variants with an AF greater than 50% were found in the observed data. Table 1 shows an overview of the identified SNVs with the highest allele frequencies.

```{r}
#| label: table-one
#| tbl-cap-location: top
#| tbl-cap: "Table 1. Variants with the highest allele frequencies."  

overlap_phred40 |>
  slice_max(AF, n = 5) |>
  select(Chrom, Pos, Ref, Alt, PHRED, "Gene" = GeneName, AF) |> 
#  kable(align = "lllllll") |>
  flextable() |>
  theme_booktabs(bold_header = TRUE) |>
  align(align = "left", part = "all") |>
  italic(j = "Gene") |> 
  flextable::colformat_num(j = 2, big.mark = "") |>
  
  #set_header_labels(Chrom = "Chromosome", Pos = "Position", 
  #          Ref = "Reference", Alt = "Alternative", PHRED = "PHRED", GeneName = "Gene", 
  #          AF = "Allele Frequency") |>
  fontsize(size = 10)

   

```

## Literature research of variants with high allele frequencies

### 1-171208951-C\>T (rs6661174)

The observed SNV with the highest allele frequncy was rs6661174. It leads to a premature stop codon on exon 9 of *FMO2*, a gene coding for Flavin Containing Dimethylaniline Monoxygenase 2. The alternative allele (T-allele) has an overall AF of `r arrange(overlap_phred40, desc(AF))$AF[1]`. It has allele frequencies of or close to 0 (100%) throughout all GnomAD ancestry groups except African / African American were the frequency is only 0.8764. The T allele leads to *FMO2* loss of function while the ancestral C allele leads to a functional allele. A study by @veeramah2008 found high frequencies of the C allele in the population of sub-Saharan Africa where approximately one third was at least heterozygote, but could not find evidence for selective pressure on either allele. In contrast, @mekonnen2017 propose that in specific African populations, the persistence of the functional allele can be explained by its association with reduced Tuberculosis disease activity.

**For Discussion: C allele does not segregate together with known TB risk alleles**

-   Problem - do the risk alleles have a molecular effect?
-   If so â€“\> might they be actually responsible for the observed reduced TB?
    -   Relevant section marked in paper

### 5-75669297-G\>A (rs34358)

This variant is a stop-gain in *ANKDD1B* , coding for Ankyrin Repeat And Death Domain Containing 1B. The alternative A allele is the major allele in all GnomAD ancestry groups except South Asian and East Asian were it has AFs of 0.44 and 0.35 respectively. The variant has been described as risk allele for type 2 diabetes and dyslipidaemia [@vujkovic2020] and as protective allele for migraine without aura [@zhang2022], however no studies investigating the evolutionary history of the allele and not evidence for a deleterious effect (like a pathogenic effect with early onset and high penetrance) could be found.

### 1-54716627-T\>A (rs1147990)

This variant has an overall frequency of 0.55. It is most frequent in the East Asian (0.98) and South Asian (0.72) ancestry group. Altough beeing annotated as stop gain in *MROH7-TTC4* by CADD v1.6, the variant is actually located in a read-trough transcript between *MROH7* and *TTC4,* two neighbouring genes. Multiple variants of the transcript can arise from alternative splicing, however they are likely destroyed by nonsense mediated decay and do not encode any protein. In CADD v1.7 which uses updated Ensemble annotations, the transcript is no longer annotated as coding gene. With that change, the variant's PHRED score was markedly reduced to 19.97.

### 7-21543345-G\>T (rs2285943)

Rs2285943 is predicted to result in a stop gain on exon 1/82 of *DNAH11.* The variant is relatively common (AFs 0.29 - 0.55) troughout all GnomAD ancestry groups. It is known that DNAH11 loss of function leads to Primary Ciliary Dyskineasia [@bartoloni2002; @lai2016; @schwabe2008], however, multiple ClinVar submissions indicated that rs2285943 is not linked to that condition.

\*\*For discussion\*\*

-   Genome Browswer shows alternative promotor just downstream of variant
-   alt promotor: <https://genome.ucsc.edu/cgi-bin/hgc?hgsid=1885897182_19ARA0SsykGacZ7HT0dy80zu9yoK&db=hg38&c=chr7&l=21543005&r=21547379&o=21543345&t=21543495&g=knownAlt&i=altPromoter>
-   Range: [chr7:21543346-21543495](https://genome.ucsc.edu/cgi-bin/hgTracks?hgsid=1885897182_19ARA0SsykGacZ7HT0dy80zu9yoK&db=hg38&position=chr7%3A21543346-21543495)

### 19-48703417-G\>A (rs601338)

The fifth of the selected variants leads to a loss of function in *FUT2.* With allele frequencies between 0.30 in Admixed Americans and 0.52 in Middle Easterns, it is common in most ancestry groups. However, the variant is rare in East Asian (AF 0.002). The variant affects the function of the alpha(1,2)fucosyltransferase which is responsible for secretion of ABO antigens. Carriers of a functional *FUT2* allele secrete these antigens in their bodily fluids while homozygous or compound heterozygous carriers fo non-functional FUT2 (for example caused by rs601338) do not secret the antigens. The secretor phenotype (secretor polymorphism) seems to be associated as both, a risk or a protective factor, with numerous conditions and diseases [@krog2023; @mottram2017].

## Enrichment analysis 
