---
title: "Gnomad Overlaps"
author: "Jonathan Speh"
format:
  html:
    code-fold: true
    code-tools: true
    df-print: kable
engine: knitr
execute: 
  cache: true
bibliography: references.bib
---

# Dependencies

```{r}
#| label: setup
#| message: false
library(data.table)
library(dplyr)
library(ggplot2)
library(tidyr)
#library(ggridges)
#library(knitr)
source("code/helpers.R")
```

# Data prepration

```{bash}
#| label: prepare-overlaps
#| eval: false

cd data/
# Download Gnomad Exome data
wget https://storage.googleapis.com/gcp-public-data--gnomad/release/4.0/vcf/exomes/gnomad.exomes.v4.0.sites.chr{1..22,X,Y}.vcf.bgz

# Select entries with pass flag
(for i in {1,{10..19},2,{20..22},{3..9},X,Y}; do zcat data/gnomad.exomes.v4.0.sites.chr${i}.vcf.bgz | awk 'BEGIN{ OFS="\t"; FS="\t"}{if ($7 == "PASS") print }' | sed s/'^chr'//g | sort -k1,1 -k2,2n -k4,4 -k5,5; done ) | gzip -c > gnomad.exomes.v4.0.sites.PASS.vcf.gz

# ClinVar pathogenic overlap with gnomAD

( zcat GRCh38_CADDv1.6_whole_genome_SNVs_inclAnno.phred40.tsv.gz | head -n 1 | awk 'BEGIN{ OFS="\t"; FS="\t" }{ print $0,"GenomAD-Exomes" }' ; join --nocheck-order -t"$(echo -e '\t')" -j 1 <(zcat clinvar_20231112.pathogenic_SNVs_CADDv1.6_anno.tsv.gz | tail -n +3 | awk 'BEGIN{ OFS="\t"; FS="\t" }{ print $1":"$2":"$3">"$4,$0 }' ) <( zcat gnomad.exomes.v4.0.sites.PASS.vcf.gz | awk 'BEGIN{ OFS="\t"; FS="\t" }{ if ((length($4) == length($5)) && (length($4) == 1)) { print $1":"$2":"$4">"$5,$8 } }' ) | cut -f 2- ) | gzip -c > clinvar_20231112.pathogenic_SNVs_CADDv1.6_anno.gnomad-exome_overlap.tsv.gz


# PHRED 40 CADD score overlap with gnomAD

( zcat GRCh38_CADDv1.6_whole_genome_SNVs_inclAnno.phred40.tsv.gz | head -n 1 | awk 'BEGIN{ OFS="\t"; FS="\t" }{ print $0,"GenomAD-Exomes" }' ; join --nocheck-order -t"$(echo -e '\t')" -j 1 <(zcat GRCh38_CADDv1.6_whole_genome_SNVs_inclAnno.phred40.tsv.gz | tail -n +2 | awk 'BEGIN{ OFS="\t"; FS="\t" }{ print $1":"$2":"$3">"$4,$0 }' ) <( zcat gnomad.exomes.v4.0.sites.PASS.vcf.gz | awk 'BEGIN{ OFS="\t"; FS="\t" }{ if ((length($4) == length($5)) && (length($4) == 1)) { print $1":"$2":"$4">"$5,$8 } }' ) | cut -f 2- ) | gzip -c > GRCh38_CADDv1.6_whole_genome_SNVs_inclAnno.phred40.gnomad-exome_overlap.tsv.gz

# ALL POTENTIAL STOP GAINED overlap gnomAD

( zcat GRCh38_CADDv1.6_whole_genome_SNVs_inclAnno.phred40.tsv.gz | head -n 1 | awk 'BEGIN{ OFS="\t"; FS="\t" }{ print $0,"GenomAD-Exomes" }' ; join --nocheck-order -t"$(echo -e '\t')" -j 1 <(zcat GRCh38_CADDv1.6_SNVs_STOP_GAINED.tsv.gz | awk 'BEGIN{ OFS="\t"; FS="\t" }{ print $1":"$2":"$3">"$4,$0 }' ) <( zcat gnomad.exomes.v4.0.sites.PASS.vcf.gz | awk 'BEGIN{ OFS="\t"; FS="\t" }{ if ((length($4) == length($5)) && (length($4) == 1)) { print $1":"$2":"$4">"$5,$8 } }' ) | cut -f 2- ) | gzip -c > GRCh38_CADDv1.6_SNVs_STOP_GAINED.gnomad-exome_overlap.tsv.gz

cd ../
```

# Overlap PHRED \> 40 - Gnomad

## Load and check file

```{r}
#| label: process-phred40-overlap
keep <- c("Chrom" = "#Chrom", "Pos", "Ref", "Alt", "Type", "Length",
         "AnnoType", "Consequence", "ConsScore", "ConsDetail", 
         "GeneID", "GeneName", "FeatureID", "Intron", "Exon", "cDNApos", 
         "relcDNApos", "protPos", "relProtPos", "RawScore", "PHRED", 
         "GnomAD_Exomes" = "GenomAD-Exomes")


overlap_phred40 <- process_caddscoring(fread("data/GRCh38_CADDv1.6_whole_genome_SNVs_inclAnno.phred40.gnomad-exome_overlap.tsv.gz"),
                                    cols = keep, 
                                    autosomes_only = TRUE) 
overlap_phred40_unique <- remove_duplicates(overlap_phred40)

# Parse Gnomad
overlap_phred40_unique_parsed <- parse_gnomad(overlap_phred40_unique)
```

Check if all Variants in `overlap_phred40_unique` are unique: `r nrow(overlap_phred40_unique) == length(unique(overlap_phred40_unique$Ident))`

Check if number of entries in `overlap_phred40_unique` equals unique ids in `overlap_phred40`: `r nrow(overlap_phred40_unique) == length(unique(overlap_phred40$Ident))`

Check if all Variants are SNVs: `r mean(overlap_phred40_unique$Type == "SNV") == 1`

Check if parsing kept IDs. intact: `r mean(overlap_phred40_unique[,1:10] == overlap_phred40_unique_parsed[,1:10]) == 1`

## Description of resulting data

A total of `r nrow(overlap_phred40)` overlaps, resulting in `r nrow(overlap_phred40_unique)` unique variants where found (excluding sex chromosomes).

All overlaps where in coding regions, and most of them where Stop-Gains

```{r}
#| label: summarise-phreds
#| tbl-cap: "Most variants with a Phred > 40 are Stop-gains" 

overlap_phred40_unique_parsed |> group_by(Consequence) |> count() |> arrange(desc(n))


overlap_phred40_unique_parsed |> 
  ggplot(aes(x = PHRED)) +
  geom_bar(stat = "count")
```

## Allele frequencies of predicted deleterious Variants found in GnomAD

```{r}
#| label: af-phred40
#| tbl-cap: "Allele frequencies of the observed variants" 
#| tbl-subcap: 
#|  -  "Most Variants have only been observed once or a few times"
#|  -  "Variants with an AC > 50"

overlap_phred40_unique_parsed |>
  group_by(AC) |> count() |> arrange(desc(n))
  
overlap_phred40_unique_parsed |> arrange(desc(AC)) |> 
  select(Chrom, Pos, Ref, Alt, GeneName, relcDNApos, AC, AF, PHRED) |> 
  filter(AC > 50) |>
  mutate(AF = format(AF, scientific = FALSE))

```

### Variants with high AC / AF

**Continue here:** - annotate Genes with hight AFs - describe top hits

# Overlap Stop Gain - Gnomad
```{r}
#| label: process-stopgain-gnomad

overlap_stopgain <- process_caddscoring(fread("data/GRCh38_CADDv1.6_SNVs_STOP_GAINED.gnomad-exome_overlap.tsv.gz"),
                                     cols = keep, autosomes_only = TRUE) 

overlap_stopgain_unique <- remove_duplicates(overlap_stopgain)

overlap_stopgain_unique_parsed <- parse_gnomad(overlap_stopgain_unique)
```

Check if all Variants in `overlap_stopgain_unique` are unique: `r nrow(overlap_stopgain_unique) == length(unique(overlap_stopgain_unique$Ident))`

Check if number of entries in `overlap_stopgain_unique` equals unique ids in `overlap_stopgain`: `r nrow(overlap_stopgain_unique) == length(unique(overlap_stopgain$Ident))`

Check if all Variants are SNVs: `r mean(overlap_stopgain_unique$Type == "SNV") == 1`

Check if all Variants are Stop Gains: `r mean(overlap_stopgain_unique$Consequence == "STOP_GAINED") == 1`

Check if parsing kept IDs. intact: `r mean(overlap_stopgain_unique[,1:10] == overlap_stopgain_unique_parsed[,1:10]) == 1`

## Description of resulting data

A total of `r nrow(overlap_stopgain)` overlaps, resulting in `r nrow(overlap_stopgain_unique)` unique variants where found (excluding sex chromosomes).

```{r}
#| label: plot-stopgain-phreds
#| fig-cap: "Distrubtion of the PHRED Scores of stop-gains in Gnomad. Vertical line indicates median PHRED"

overlap_stopgain_unique_parsed |>
  ggplot(aes(x = PHRED)) +
  geom_density() +
  geom_vline(aes(xintercept = median(PHRED)), linetype = 2)  


overlap_stopgain_unique_parsed |> head()
  
```




## Allele frequencies
```{r}
#| label: plot-stopgain-af
#| tbl-cap: "Allele frequencies of the stopgains that were observed in Gnomad" 
#| tbl-subcap: 
#|  -  "Most Variants have only been observed once or a few times"
#|  -  "Variants with an AC > 50"
#| fig-cap: 
#|   -  "There is no appearant linear relationship between AF and PHRED in the observed stop-gains" 
#|   -  "Distribution of PHRED in variants with AC below and above 50"

overlap_stopgain_unique_parsed |>
  ggplot(aes(x = PHRED, y = AF)) +
  geom_point() +
  scale_y_log10()

# TODO make ridgeplot,, AC 50 is quite arbitrary
overlap_stopgain_unique_parsed |> 
  mutate(AC_50 = ifelse(AC > 50, TRUE, FALSE)) |>
  ggplot(aes(x = PHRED, colour = AC_50, fill = AC_50)) +
  geom_density(alpha = 0.1)

overlap_stopgain_unique_parsed |>
  group_by(AC) |> count() |> arrange(desc(n))
  
overlap_stopgain_AC50 <- overlap_stopgain_unique_parsed |> arrange(desc(AC)) |> 
  select(Chrom, Pos, Ref, Alt, GeneName, relcDNApos, AC, AF, PHRED) |> 
  filter(AC > 50)

mutate(overlap_stopgain_AC50, AF = format(AF, scientific = FALSE))
```


## relative Position in gene

**TODO:**
-  Write all_positions to disc and reload in new chunk
-  Plot distribution of relcDNApos of all theoretical vs. observed stopgains


```{r}
#| label: plot-rel-pos
all_stopgains <- fread("data/GRCh38_CADDv1.6_SNVs_STOP_GAINED.tsv.gz")
colnames(all_stopgains) <- fread("data/CADD_columns.txt")$V2
all_stopgains <- process_caddscoring(all_stopgains, 
                                     cols = c("Chrom" = "#Chrom", "Pos", "Ref", 
                                              "Alt", "Type", "Length", "AnnoType", 
                                              "Consequence", "ConsScore", 
                                              "GeneName", "Exon", 
                                              "cDNApos", "relcDNApos", "protPos", 
                                              "relProtPos", "RawScore", "PHRED"),
                                     autosomes_only = TRUE)

all_stopgains_unique <- remove_duplicates(all_stopgains)
all_stopgains_unique <- all_stopgains_unique |> group_by(Ident) |> 
  slice_sample(n = 1)

# TODO - continue here with writing to disc

observed_stopgain_positions <- overlap_stopgain_unique_parsed$relcDNApos
all_stopgain_positions <- all_stopgains_unique$relcDNApos
sample_stopgain_positions <- sample(all_stopgain_positions, length(observed_stopgain_position), 
                                    replace = FALSE)


positions <- data.frame(position = c(observed_stopgain_position, all_stopgain_position),
                        flag = c(rep("observed", length(observed_stopgain_position)),
                                 rep("simulated", length(all_stopgain_position))))

positions |> 
  ggplot(aes(x = position, colour = flag, fill = flag)) +
  geom_density(alpha = 0.1)


```












